### DISCUSSION TITLE
import com.getcapacitor.android.R;

### DISCUSSION THREAD

**Discussion by Axmon** [ionic-team/capacitor#6521] [Q&A]
Hi, I'm new to capacitor and wanted to try to build an app with android studio.
My build failed due to an import error: import com.getcapacitor.android.R;

Now I'm checking the git repro because I thought soemthing went wrong with my package imports.

In v4.7.3 capacitor/android/capacitor/src/main/java/com/getcapacitor/
the following classes have the import com.getcapacitor.android.R;
Bridge.java, BridgeActivity.java, BridgeFragment.java

I can't find the package com.getcapacitor.android.R

May somebody help me with this issue.

Thanks a lot in advance!

**Comment by Axmon** [comment]
Tried capacitor v3.9.0 the package import com.getcapacitor.android.R; is found.
Is this propaply an issue with my android studio config?
Or already in my vue package config?
Before I follwed: https://capacitorjs.com/docs/getting-started

Switched to 
npm i @capacitor/core@latest-3
npm install @capacitor/cli --save-dev
npm i @capacitor/android@latest-3 @capacitor/ios@latest-3

and the test App is build in android studio
Flamingo 2022.2.1
Build #AI-222.4459.24.2221.9862592, built on March 31, 2023
Runtime version: 17.0.6+0-17.0.6b802.4-9586694 amd64
VM: OpenJDK 64-Bit Server VM by JetBrains s.r.o.

**Comment by Jaredbeyers23** [comment]
Any updates that causing the issue? Having the same problem

**Comment by raul-ruiz-gallardo** [comment]
Same problem here with capacitor 6.

**Reply by archanag2024** [reply]
Hi
Did you find a solution for this?

**Reply by raul-ruiz-gallardo** [reply]
Sorry for this late response. In my case, the problem was that I was trying to use the bridge to registerForActivityResult in order to request permissions. Now, I'm using the Plugin.requestPermissionForAlias method that accepts a callback method name that will be hit when the permission request is finished. Again, in my case I'm using the default checkPermissions method, but in your plugin you can override this method.

**Reply by raul-ruiz-gallardo** [reply]
Anyway, android studio keeps marking the import inside the bridge.java file as invalid, but I'm not importing Bridge in my plugin file.

**Comment by zcattacz** [comment]
In my case, it turned out 
- These must match
`
  @capacitor/ios: not installed
  @capacitor/core: 6.2.0
  @capacitor/cli: 6.2.0
  @capacitor/android: 6.2.0
`
Previously the cap/core/cli is @6.0.0 and cap/android is @6.2.0.
`npx cap doctor` concluded `Android looks great`, despite everything blew up:
 from `circular reference in dependency`, `androidx.xxx is missing` to this `com.getcapacitor.android.R` is missing. They just keep popping up one after another,
until I zap, recreated the cap project, and made sure these three pkg are of the same version before opening Android Studio. Then it just compiles in one go.
- If you are using capacitor6, with latest studio, `JDK_HOME` and gradle JDK version should be set JDK17. If you have already tried to build the project, better empty the cap folder and start over with correct setting before trying to build the project. The `clear cache` studio command doesn't always help.
